schloss/producer.py

```
@@ -0,0 +1,53 @@
+import asyncio
+import logging
+from socket import gaierror
+from typing import Optional
+
+import aiokafka
+from kafka.errors import KafkaConnectionError
+
+logger = logging.getLogger(__name__)
+
+
+class SchlossProducer:
+    def __init__(self, url, **kwargs):
+        loop = asyncio.get_running_loop()
+        self.producer = aiokafka.AIOKafkaProducer(
+            loop=loop, bootstrap_servers=url, **kwargs
+        )
+        self._start_task = None
+
+    async def start(self):
+        if self._start_task is not None:
+            raise ValueError('Producer is already started')
+        self._start_task = asyncio.create_task(self._start())
+
+    async def _start(self):
+        timeout = 2
+        while True:
+            try:
+                await self.producer.start()
+            except (KafkaConnectionError, gaierror):
```

Other exceptions will be silently lost (or shown at shutdown)?

---

Ah, I see. It's awaited in `send()`. I've not sure it's fine, have to sleep on it.